src/openrct2/scenario/ScenarioRepository.cpp

```diff
@@ -272,6 +275,35 @@ class ScenarioRepository final : public IScenarioRepository
         delete scanner;
     }
 
+    void ScanMegaPark(std::string &mpdatDir, std::string &scenarioDir)
+    {
+        //Convert mp.dat from RCT1 Data directory into SC21.sc4 (Mega Park)
+        utf8 mpdatPath[MAX_PATH];
+        utf8 sc21Path[MAX_PATH];
+
+        String::Set(mpdatPath, sizeof(mpdatPath), mpdatDir.c_str());
+
+        String::Set(sc21Path, sizeof(sc21Path), scenarioDir.c_str());
+        //Ensure the scenario directory exists first
+        platform_ensure_directory_exists(sc21Path); 
+        Path::Append(sc21Path, sizeof(sc21Path), "SC21.SC4");
+
+        if (platform_file_exists(mpdatPath) && !(platform_file_exists(sc21Path)))
+        {
+            size_t length;
+            auto mpdat = (uint8 *)(File::ReadAllBytes(mpdatPath, &length));
```

Comment:
Oh, `File::ReadAllBytes` returns `void *`. I'd say that's a bit odd, given the name of this function (`Bytes`) and I would've expected it to return `uint8 *` already. @IntelOrca ?

You should probably call `Memory::FreeArray` on that pointer once done with it.